<launch>

    <!--args for apritag_ros-->
    <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
    <arg name="node_namespace" default="apriltag_ros_continuous_node" />
    <arg name="camera_name" default="/airsim/rgb" />
    <arg name="camera_frame" default="front_center_custom_optical" />
    <arg name="image_topic" default="image_rect_color" />
    <arg name="wsl_ip" default="172.28.224.1"/>

    <!--Run image_raw-->
    <node pkg="airsim_ros_pkgs" type="image_raw.py" name="image_raw"> 
        <param name="wsl_ip" value="$(arg wsl_ip)"/>
    </node>
    
    <!--Run image proc-->
    <node 
    pkg="image_proc" 
    type="image_proc" 
    name="image_proc" 
    ns="/airsim/rgb">
    </node>

    <!--Run apriltag ros-->
    <include file="$(find apriltag_ros)/launch/continuous_detection.launch">
        <arg name="launch_prefix" value="$(arg launch_prefix)" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
        <arg name="node_namespace" value="$(arg node_namespace)" />
        <arg name="camera_name" value="$(arg camera_name)" />
        <arg name="camera_frame" value="$(arg camera_frame)" />
        <arg name="image_topic" value="$(arg image_topic)" />
	</include>

    <!--ekf apriltag estimation-->
    <node 
    pkg="utm"
    type="ekf_tag.py"
    name="ekf_tag"/>
    
</launch>       